bubuku/features/swap_partitions.py

```diff
@@ -0,0 +1,180 @@
+from collections import namedtuple
+import logging
+from operator import attrgetter
+
+from bubuku.broker import BrokerManager
+from bubuku.controller import Check
+from bubuku.features.rebalance import BaseRebalanceChange
+from bubuku.zookeeper import BukuExhibitor
+
+_LOG = logging.getLogger('bubuku.features.swap_partitions')
+
+TpData = namedtuple('TpData', ('topic', 'partition', 'size', 'replicas'))
+
+
+class SwapPartitionsChange(BaseRebalanceChange):
+    def __init__(self, zk: BukuExhibitor, fat_broker_id: int, slim_broker_id: int, gap: int, size_stats: dict):
+        self.zk = zk
+        self.fat_broker_id = fat_broker_id
+        self.slim_broker_id = slim_broker_id
+        self.gap = gap
+        self.size_stats = size_stats
+        self.to_move = None
+
+    def run(self, current_actions):
+        if self.should_be_cancelled(current_actions):
+            _LOG.info("Cancelling swap partitions change as there are conflicting actions: {}".format(current_actions))
+            return False
+        try:
```

Comment:
I think that it should be checked if there is rebalance in progress. If data is still moving, it's better to wait, because partition distribution information will be incorrect.